lib/indexeddb.js

```diff
@@ -0,0 +1,118 @@
+// Implemented by window & worker
+declare interface IDBEnvironment {
+  indexedDB: IDBFactory;
+}
+
+type IDBDirection = 'next' | 'nextunique' | 'prev' | 'prevunique';
+
+// Implemented by window.indexedDB & worker.indexedDB
+declare interface IDBFactory {
+  open(name: string, version?: number): IDBOpenDBRequest;
+  deleteDatabase(name: string): IDBOpenDBRequest;
+  cmp(a: any, b: any): -1|0|1;
+}
+
+declare interface IDBRequest extends EventTarget {
+  result: IDBObjectStore;
+  error: Error;
+  source: ?(IDBIndex | IDBObjectStore | IDBCursor);
+  transaction: IDBTransaction;
+  readyState: 'pending'|'done';
+  onerror: (err: any) => void;
```

Comment:
Nowadays we're mostly writing our callbacks as `(err: any) => mixed`. People really like their arrow functions, and the `void` return type was getting in the way.